logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
@jakelandis simulation :D 

```java
    @Test
    public void murmurCollisions() {
        int i = 0;
        final Random random = new SecureRandom();
        final int count = 60;
        final Set<Integer> hashes = new HashSet<>(count * 2);
        while (true) {
            for (int j = 0; j < count; ++j) {
                hashes.add(random.nextInt());
            }
            if (hashes.size() < count) {
                System.out.println("collision");
            }
            hashes.clear();
            ++i;
            if (i % 1_000_000 == 0) {
                System.out.println("million");
                i = 0;
            }
        }
    }
```

(best implementation I could code in 2 min :D) 

Prints:

```sh
million
collision
million
million
million
million
```

It's just the law of large numbers at play here